contracts/interchain-token-service/src/testutils.rs

```diff
@@ -1,2 +1,160 @@
+) -> InterchainTokenServiceClient<'a> {
+    let owner = Address::generate(env);
+    let operator = Address::generate(env);
+    let its_hub_address = String::from_str(env, "its_hub_address");
+    let chain_name = String::from_str(env, "chain_name");
+
+    // Note: On changes to `interchain-token` crate, recompile it via `stellar contract build && ./optimize.sh`
+    // and copy the built `target/wasm32-unknown-unknown/release/interchain_token.optimized.wasm` to ../testdata.
+    let interchain_token_wasm_hash = env
+        .deployer()
+        .upload_contract_wasm(INTERCHAIN_TOKEN_WASM_HASH);
+
+    let native_token_address = env.register_stellar_asset_contract_v2(owner.clone());
+
+    let contract_id = env.register(
+        InterchainTokenService,
+        (
+            &owner,
+            &operator,
+            &gateway.address,
+            &gas_service.address,
+            its_hub_address,
+            chain_name,
+            native_token_address.address(),
+            interchain_token_wasm_hash,
+        ),
+    );
+
+    InterchainTokenServiceClient::new(env, &contract_id)
+}
+
+#[cfg(any(test, feature = "testutils"))]
+pub fn setup_env<'a>() -> (
+    Env,
+    InterchainTokenServiceClient<'a>,
+    AxelarGatewayClient<'a>,
+    AxelarGasServiceClient<'a>,
+    TestSignerSet,
+) {
+    let env = Env::default();
+
+    let (signers, gateway_client) = setup_gateway(&env, 0, 5);
+    let gas_service_client: AxelarGasServiceClient<'_> = setup_gas_service(&env);
+
+    let client = setup_its(&env, &gateway_client, &gas_service_client);
+
+    (env, client, gateway_client, gas_service_client, signers)
+}
+
+#[cfg(any(test, feature = "testutils"))]
+#[allow(dead_code)]
+pub fn setup_gas_token(env: &Env, sender: &Address) -> Token {
+    let asset = &env.register_stellar_asset_contract_v2(Address::generate(env));
+    let gas_amount: i128 = 1;
+    let gas_token = Token {
+        address: asset.address(),
+        amount: gas_amount,
+    };
+
+    StellarAssetClient::new(env, &asset.address())
+        .mock_all_auths()
+        .mint(sender, &gas_amount);
+
+    gas_token
+}
+
+#[cfg(any(test, feature = "testutils"))]
+#[allow(dead_code)]
+pub fn setup_its_token(
+    env: &Env,
+    client: &InterchainTokenServiceClient,
+    sender: &Address,
+    supply: i128,
+) -> BytesN<32> {
+    let salt = BytesN::from_array(env, &[1u8; 32]);
+    let token_metadata = TokenMetadata {
+        name: String::from_str(env, "Test"),
+        symbol: String::from_str(env, "TEST"),
+        decimal: 18,
+    };
+
+    let token_id = client.mock_all_auths().deploy_interchain_token(
+        sender,
+        &salt,
+        &token_metadata,
+        &supply,
+        &None,
+    );
+
+    token_id
+}
+
+#[cfg(any(test, feature = "testutils"))]
+#[allow(dead_code)]
+pub fn register_chains(_env: &Env, client: &InterchainTokenServiceClient) {
+    client
+        .mock_all_auths()
+        .set_trusted_chain(&client.its_hub_chain_name());
```

Comment:
should remove the usage of `register_chains` completely since it doesn't make much sense. Do something like this instead
https://github.com/axelarnetwork/axelar-cgp-soroban/blob/f52ddede4fef910361a306d47b2f248cd3897923/contracts/interchain-token-service/tests/execute.rs#L87